plugins/become/doas.py

```diff
@@ -111,17 +111,18 @@ def build_become_command(self, cmd, shell):
 
         self.prompt = True
 
-        become_exe = self.get_option('become_exe')
+        become_exe = self.get_option('become_exe') or '/usr/bin/doas'
```

Comment:
This is not the correct fix. First, the default value is `doas`, not `/usr/bin/doas` (see argument spec). Second, this should not be necessary if the option processing mechanism is working correctly. So either this plugin is doing something wrong, or there's a bug in Ansible.

Reply:
it was `self.name` AFAIR before, I can restore that.

Reply:
No, that's completely wrong, since it is `community.general.doas`. My PR originally replaced that with `doas` IIRC, but I was told in review that that's wrong and `self.get_option` should handle this.

Reply:
Ok, put `doas`. Since this is OpenBSD, and the path of `doas` is not supposed to change. I though it was more secure to use the full path.

Reply:
Thanks for the clarification regarding `get_option()`, I didn't know it was supposed to read the documentation keys. I just tried with different versions of Ansible:

- 2.9: ok
- 2.10: ko
- devel: ko

Reply:
BTW, with 2.9 you need to explicitly write `become_method: community.general.doas`, otherwise you end up using 2.9's copy.